fluent-bundle/src/bundle.rs

```diff
@@ -158,27 +159,37 @@ impl<R> FluentBundle<R> {
             .into_iter()
             .map(|s| s.clone().into())
             .collect::<Vec<_>>();
-        let default_langid = "en".parse().expect("Failed to parse `en` langid.");
-        let pr_locale = negotiate_languages(
-            &locales,
-            &PluralRules::get_locales(PluralRuleType::CARDINAL),
-            Some(&default_langid),
-            NegotiationStrategy::Lookup,
-        )[0]
-        .clone();
+        let lang = locales.get(0).expect("Expect at least one language").clone();
+        FluentBundle {
+            locales,
+            resources: vec![],
+            entries: HashMap::new(),
+            intls: Rc::new(RefCell::new(IntlLangMemoizer::new(lang))),
```

Comment:
I'm not sure if I like the fact that the lang memoizer is keyed on single language codes. _The_ reason why `FluentBundle::new` takes a vector of locales as argument is to be able to offer better negotiation results from intl formatters.

If keying on the lang is a necessity, perhaps consider parsing the lang code first, so that Unicode extensions are stripped, etc?

Reply:
> I'm not sure if I like the fact that the lang memoizer is keyed on single language codes. The reason why FluentBundle::new takes a vector of locales as argument is to be able to offer better negotiation results from intl formatters.

Correct, but in Rust case, we so far did not include negotiation as part of formatter constructor, but rather as an external operation.

In other words, we expect the locale provided to the constructor to be available, or the constructor will return an error.

This allows the callee to use any negotiation strategy they seem fit.

Reply:
We so far have not been enabling unicode extension keys to affect the formatters. This will change in the future with `unic-locale` replacing `unic-langid`.

Reply:
> In other words, we expect the locale provided to the constructor to be available, or the constructor will return an error.

Do you mean the constructor for `FluentBundle`, or the formatter?

Reply:
FluentBundle. Basically, I'd expect there to be a moment where we do:

```rust
let pluralLocales = PluralRules::getAvailableLocales();
let datetimeLocales = DateTimeFormat::getAvailableLocales();
let numberLocales = NumberFormat::getAvailableLocales();
let fluentLocales = getL10nLocales();
let supportedLocales = negotiateLocales(dateTimeLocales, pluralLocales, numberLocales, fluentLocales);

let bundle = FluentBundle::create(supportedLocales);
```

Reply:
Is there any reason left then to pass `supportedLocales` rather than the first `supportedLocale`? Should we change it?

Reply:
For now, I planned to do this negotiation from within of the constructor, but with the external formatters which should be considered for the negotiation, it may not be possible after all... so maybe it makes sense to just pass a single locale to the bundle?

Reply:
@Pike Do you have more thoughts about this?

Reply:
May also be related - https://github.com/unicode-org/message-format-wg/issues/3#issuecomment-574777459 :)